app/controllers/users/webauthn_setup_controller.rb

```diff
@@ -191,10 +191,8 @@ def process_invalid_webauthn(form)
         else
           flash.now[:error] = t('errors.webauthn_setup.unique_name')
         end
-      elsif form.platform_authenticator?
-        flash[:error] = t('errors.webauthn_platform_setup.general_error')
```

Comment:
I looked a little closer at this since I was worried about potentially showing users messaging about security keys when enrolling face or touch unlock. It seems like this path can be reached, but only in some rare cases where the form data becomes corrupt for some reason, triggering one of the [validation behaviors in `WebauthnSetupForm`](https://github.com/18F/identity-idp/blob/cfbf165d575d280b0ee6f001d71e1db9e8165546/app/forms/webauthn_setup_form.rb#L4-L8). For example, I was able to produce the following screenshot by manipulating the name of one of the form fields (`name="attestation_object"` to `name="foo"`).

![image](https://github.com/18F/identity-idp/assets/1779930/d0c0a88f-4f46-488d-bdb8-81858f2908c3)

A more real-world scenario might be encountered if the [attestation object validation](https://github.com/18F/identity-idp/blob/cfbf165d575d280b0ee6f001d71e1db9e8165546/app/forms/webauthn_setup_form.rb#L82-L97) fails for some reason, e.g. bad data from the device.

With that in mind, I think we probably want to keep this messaging in place, since we definitely don't want to show text about "security keys" on this setup screen. We could either propose to cancel the ticket, or use it as an opportunity to refactor.

For refactoring, specifically I still find it odd how the form class will validate and add error messages to its `errors` property, but then those errors are not used from the controller class itself. Ideally, the controller would pass through the errors from the form class. This is made slightly more difficult by the fact that we've probably come to _rely_ on those errors not being passed through, and specific error messages might not be very user-friendly (especially the [attestation error message](https://github.com/18F/identity-idp/blob/cfbf165d575d280b0ee6f001d71e1db9e8165546/config/locales/errors/en.yml#L121-L124) produced from [the aforementioned validation](https://github.com/18F/identity-idp/blob/cfbf165d575d280b0ee6f001d71e1db9e8165546/app/forms/webauthn_setup_form.rb#L82-L97)). In that case, I'd either suggest getting some UX feedback, or replacing those undesirable messages with our blanket "unable to add" message.

Reply:
Do you recommend cancelling this one then creating a follow-up ticket concerning refactoring?

Reply:
> Do you recommend cancelling this one then creating a follow-up ticket concerning refactoring?

I could go either way. I think closing this one could work well to wrap up the the ticket before end-of-sprint, but I also think it's reasonable to fit the refactoring as part of the existing ticket, or at least the administrative overhead of creating a separate ticket will likely delay and add extra effort.

I'll leave it up to you! Could you write up that ticket if you decide to go that route?